src/DogStatsd.php

```diff
@@ -90,6 +96,12 @@ public function __construct(array $config = array())
 
         $this->socketPath = isset($config['socket_path']) ? $config['socket_path'] : null;
 
+        $this->transportFactory = (
+            isset($config['transport_factory']) ?
+            $transportFactory :
+            new BridgingTransportFactory($this->socketPath, $this->host, $this->port)
+        );
+
```

Comment:
What I had hoped might be pointed out is that this could be an alternative to https://github.com/DataDog/php-datadogstatsd/pull/116/files#diff-baf3cb376e36dfcf767f6389163de7d731798b1b108a44f33a2047e8e98bae3bR88-R98

I didn't want to leap too far in a single refactoring; and I think this should be part of a Configuration, that specific factory implementations could offer. So that implementations of transport are simple, and do not have cascading logic, or magic coming from the environment; but another layer can add that for them, or omit that, as the example in the README.md does.